modules/auxiliary/client/telegram/send_message.rb

```diff
@@ -0,0 +1,55 @@
+# Content: Metasploit Module to send telegram message
+# Author: Ege Balcı| @egeblc | https://pentest.blog
+# Date: 07/2020
+
+require 'uri'
+
+class MetasploitModule < Msf::Auxiliary
+    def initialize
+        super(
+        'Name' => 'Telegram Message Client',
+        'Description' => 'This Module will send a telegram message to given chat ID with given bot token',
+        'Author' => 'Ege Balcı <egebalci@pm.me>',
+        'License' => MSF_LICENSE,
+        )
+
+        register_options(
+            [
+                OptString.new('BOT_TOKEN', [true, 'Telegram BOT token', '']),
+                OptInt.new('CHAT_ID', [true, 'Chat ID for the BOT', '']),
+                OptString.new('MSG', [true, 'Message content', 'New session opened !']),
+                OptString.new('FORMATTING', [false, 'Set message formating option (Markdown|MarkdownV2|HTML)', 'Markdown']),
+            ], self.class
+        )
+    end
+
+    def message
+        datastore['MSG']
+    end
+
+    def formatting
+        datastore['FORMATTING']
+    end
+
+    def bot_token
+        datastore['BOT_TOKEN']
+    end
+
+    def run
+        unless formatting == 'Markdown' || formatting == 'MarkdownV2' || formatting == 'HTML'
+            raise 'Invalid formatting selected'
```

Comment:
It would be better to use `fail_with(Failure::BadConfig, 'Invalid formatting selected!')` here, as this will allow for better formatted output and this is the standard way to throw configuration errors within Metasploit.

Reply:
This was resolved by using `OptEnum.New` instead and forcing the `FORMATTING` option to be required, thus removing the need for this code.